src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,181 @@
+    AdministrateurDTO administrateurDTO = null;
+    Long id;
+    boolean isAdmin = false;
+    try {
+      proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
+      id = proprietaireDTO.getId();
+    } catch (Exception e) {
+      try {
+        administrateurDTO = administrateurService.getAdministrateurByEmail(email);
+        id = administrateurDTO.getId();
+        isAdmin = true;
+      } catch (Exception exception) {
+        throw new UtilisateurNotFoundException();
+      }
 
-    private final TokenCacheService tokenCacheService;
-
-    public AuthService(ProprietaireService proprietaireService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
-        this.proprietaireService = proprietaireService;
-        this.jwtHelper = jwtHelper;
-        this.tokenCacheService = tokenCacheService;
     }
 
-    public AuthLoginResponseDTO loginProprietaire(@NonNull String email, @NonNull String password){
-        ProprietaireDTO proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
-
-        if(!proprietaireService.verifyPassword(password, proprietaireDTO.getId())){
-            throw new ProprietaireNotFoundException();
-        }
-
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if (uuid == null){
-            uuid = UUID.randomUUID().toString();
-            tokenCacheService.addToCache(proprietaireDTO.getId() ,uuid);
-        }
-
-        String token = jwtHelper.generateToken(email, uuid);
-
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    if (proprietaireDTO != null && !proprietaireService.verifyPassword(password,
+        proprietaireDTO.getId())) {
+      throw new ProprietaireNotFoundException();
+    } else if (administrateurDTO != null && !administrateurService.verifyPassword(password,
+        administrateurDTO.getId())) {
+      throw new AdministrateurNotFoundException();
     }
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    String uuid = tokenCacheService.getFromCache(id);
+    if (uuid == null) {
+      uuid = UUID.randomUUID().toString();
+      tokenCacheService.addToCache(id, uuid);
     }
 
-    public void logoutProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
+    String token = jwtHelper.generateToken(email, uuid, isAdmin);
+
+    return new AuthLoginResponseDTO(id, token, isAdmin);
+  }
+
+  /**
+   * Authentifie un utilisateur à partir d'un token.
+   *
+   * @param token Le token JWT fourni à l'utilisateur après une connexion réussie.
+   * @return Un objet {@link AuthLoginResponseDTO} contenant l'ID de l'utilisateur, le token, et si
+   * l'utilisateur est un administrateur.
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagContinuationIndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line continuation have incorrect indentation level, expected level should be 4.